File: lista-de-exercicios/exercicio-7/Fibonacci.ts
export class Fibonacci {
    private numero: number;

    public constructor (numero: number) {
        this.numero = numero;
    }

    public serializar(): Array<number> {
        let serie: Array<number> = [];
        let f0 = 0;
        let f1 = 1;
        
        if (Number.isInteger(this.numero) && this.numero >= 0) {
            serie.push(f0)
            if (this.numero > 0) {
                serie.push(f1)
                for (let i = 2; i <= this.numero; i ++) {
                    let f = f0 + f1;
                    serie.push(f);
                    f0 = f1;
                    f1 = f;
                }
            } 
        } else
            throw Error ('Número inválido');
        return serie;
    }
}

File: lista-de-exercicios-java/src/main/java/org/example/exercicio07/Fibonacci.java
package org.example.exercicio07;

import java.util.ArrayList;

public class Fibonacci {

    public ArrayList<Integer> serializar(int numero) {
        ArrayList<Integer> serie = new ArrayList<>();
        int f = 0;

        if (numero >= 0) {
            for (int i = 0; i <= numero; i++) {
                serie.add(f);
                int proximo;
                if (i == 0)
                    proximo = 1;
                else
                    proximo = serie.get(i - 1) + f;
                f = proximo;
            }
        } else {
            throw new IllegalArgumentException("Número inválido");
        }
        return serie;
    }
}
